test/test.js

```diff
@@ -360,10 +383,11 @@ describe('Routes', () => {
   })
   describe('GET /retrievals/:id', () => {
     it('gets a fresh retrieval', async () => {
-      const createRequest = await fetch(
-        `${spark}/retrievals`,
-        { method: 'POST' }
-      )
+      const createRequest = await fetch(`${spark}/retrievals`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ sparkVersion: '1.2.3' })
```

Comment:
Do you have a test to verify that `sparkVersion` is not required?

It would be nicer if you could keep this test with no changes to prove that you are not introducing any regressions and then add a new test focused on handling the new field `sparkVersion`.

The new test can be super simple, as we don't care about any request/response fields  except `sparkVersion`.

Also, do you have a test to verify that `sparkVersion` is persisted in the database? The best place for this is IMO in the new test I described above.

I know that we have different views on how much tests to write. Please pick the parts of my comment that you find useful and feel free to ignore the rest.

Reply:
> Do you have a test to verify that sparkVersion is not required?

Yes, this is the `POST /retrieval > creates a retrieval` test.

> It would be nicer if you could keep this test with no changes to prove that you are not introducing any regressions and then add a new test focused on handling the new field sparkVersion.

This is covered by the `POST /retrieval` tests already, here we're not testing whether we can create proper retrievals, but whether we can read them correctly.

> Also, do you have a test to verify that sparkVersion is persisted in the database? The best place for this is IMO in the new test I described above.

Yes, this is the `POST /retrievals > handles versions` test.